### PR TITLE
Copter: Clarify calculations in get_pilot_desired_lean_angles

### PR DESCRIPTION
**Pull request by lthall**
These changes attempt to clarify what is happening with the math here and address: #20311 #20244 #20242

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -397,38 +397,41 @@ void Copter::notify_flight_mode() {
 
 // get_pilot_desired_angle - transform pilot's roll or pitch input into a desired lean angle
 // returns desired angle in centi-degrees
-void Mode::get_pilot_desired_lean_angles(float &roll_out, float &pitch_out, float angle_max, float angle_limit) const
+void Mode::get_pilot_desired_lean_angles(float &roll_out_cd, float &pitch_out_cd, float angle_max_cd, float angle_limit_cd) const
```
Comment (rmackay9) [inline ArduCopter/mode.cpp:400]: we should fix the arguments in the .h file as well..

Comment (lthall) [reply ArduCopter/mode.cpp:400]: That was stupid.

**Review by rmackay9 — APPROVED** [review]

File: ArduCopter/mode.cpp [hunk 2]
```diff
@@ -397,38 +397,41 @@ void Copter::notify_flight_mode() {
...
 
-    // do lateral tilt to euler roll conversion
-    roll_out = (18000/M_PI) * atanf(cosf(pitch_out*(M_PI/18000))*tanf(roll_out*(M_PI/18000)));
+    // thrust_angle_x and thrust_angle_y represents a level body frame thrust vector in the
+    // direction of [thrust_angle_x, thrust_angle_y] and a magnitude
+    // tan(mag([thrust_angle_x, thrust_angle_y])) * 9.81 * aircraft mass.
 
-    // roll_out and pitch_out are returned
+    // Conversion from angular thrust vector to euler angles.
+    roll_out_cd = (18000/M_PI) * atanf(cosf(thrust_angle_x_cd*(M_PI/18000))*tanf(thrust_angle_y_cd*(M_PI/18000)));
```
Comment (rmackay9) [inline ArduCopter/mode.cpp:433]: so just to be clear, the thrust_angle_x_cd will be the opposite sign from what it was before this change.  Maybe the sign doesn't matter but just want to be sure.

Comment (lthall) [reply ArduCopter/mode.cpp:433]: To get the orientation of thrust_angle_x_cd correct I add a negative sign on line 409.

This is used on lines 417, 421, 424, 433 and 434.

417 is only scaling thrust_angle_x_cd 
421 squares thrust_angle_x_cd  and therefore the sign does not matter
424 again only scales thrust_angle_x_cd 
433 takes the cos(thrust_angle_x_cd ). Given cos(-angle) = cos(angle) the sign does not matter.
finally on line 434 we set pitch_out_cd = - thrust_angle_x_cd; where we multiply again by -1 cancelling our initial multiplication by -1.

**Comment by rmackay9** [comment]
ok txs, merged!

**Pull request merged**